dagshub/data_engine/model/query_result.py

```diff
@@ -227,6 +229,43 @@ def keypairs(keys):
         else:
             raise ValueError("supported flavors are torch|tensorflow|<torch.utils.data.Dataset>|<tf.data.Dataset>")
 
+    def as_hf_dataset(self, download_datapoints=True, download_blobs=True):
+        """
+        Loads this QueryResult as a HuggingFace dataset.
+
+        The paths of the downloads are set to the local paths in the filesystem, so they can be used with
+        a ``cast_column()`` function later.
+
+        Args:
+            download_datapoints: If set to ``True`` (default), downloads the datapoint files and sets the path column\
+                to the path of the datapoint in the filesystem
+            download_blobs: If set to ``True`` (default), downloads all blob fields and sets the respective column\
+                to the path of the file in the filesystem.
+        """
+        if download_blobs:
+            # Download blobs as paths, so later a user can apply ds.cast_column on the blobs
+            self.get_blob_fields(load_into_memory=False, path_format="str")
+
+        df = self.dataframe
+
+        if download_datapoints:
+            # Do the same for the actual datapoint files, changing the path
+            new_paths = []
+            self.download_files()
+            for dp in df["path"]:
+                new_paths.append(
+                    str(self.datasource.default_dataset_location / self.datasource.source.source_prefix / dp)
```

Comment:
Could be nice to let the user specify a non default download path and pass it on to `download_files`

Reply:
Done